backend/requirements.txt

```diff
@@ -34,34 +41,52 @@ langchain-text-splitters==0.3.0
 langgraph==0.2.34
 langgraph-checkpoint==2.0.0
 langsmith==0.1.129
+markdown-it-py==3.0.0
 MarkupSafe==2.1.5
 marshmallow==3.21.3
+mdurl==0.1.2
 msgpack==1.1.0
 multidict==6.0.5
 mypy-extensions==1.0.0
 numpy==1.26.4
 ollama==0.3.3
 orjson==3.10.7
 packaging==24.1
+pandas==2.2.3
+pillow==11.0.0
```

Comment:
Consider upgrading pandas and pillow to the latest versions for better performance.